mlx_graphs/data/data.py

```diff
@@ -188,3 +188,355 @@ def is_undirected(self) -> bool:
+        """
+        if self.node_features_dict is not None:
+            return {
+                node_type: self._num_classes("node", node_type)
+                for node_type in self.node_features_dict.keys()
+            }
+        return None
+
+    @property
+    def num_edge_classes(self) -> dict[str, int]:
+        """
+        Returns a dictionary of the number of edge classes
+        for each edge type in the current graph.
+        """
+        return {
+            edge_type: self._num_classes("edge", edge_type)
+            for edge_type in self.edge_index_dict.keys()
+        }
+
+    @property
+    def num_node_features(self) -> dict[str, int]:
+        """Returns a dictionary of the number of node features for each node type."""
+        num_node_features_dict = {}
+        if self.node_features_dict is not None:
+            for node_type, node_features in self.node_features_dict.items():
+                num_node_features_dict[node_type] = (
+                    1 if node_features.ndim == 1 else node_features.shape[-1]
+                )
+        return num_node_features_dict
+
+    @property
+    def num_edge_features(self) -> dict[str, int]:
+        """Returns a dictionary of the number of edge features for each edge type."""
+        num_edge_features_dict = {}
+        if self.edge_features_dict is not None:
+            for edge_type, edge_features in self.edge_features_dict.items():
+                num_edge_features_dict[edge_type] = (
+                    1 if edge_features.ndim == 1 else edge_features.shape[-1]
+                )
+        return num_edge_features_dict
+
+    def _num_classes(self, task: Literal["node", "edge"], type_key: str) -> int:
+        labels_dict = getattr(self, f"{task}_labels_dict")
+        if labels_dict is None or type_key not in labels_dict:
+            return 0
+        labels = labels_dict[type_key]
+        if labels.size == labels.shape[0]:
+            return np.unique(np.array(labels)).size
+        return labels.shape[-1]
+
+    def __cat_dim__(self, key: str, *args, **kwargs) -> int:
+        """
+        This method can be overriden when batching is used with custom
+        attributes. Given the name of a custom attribute `key`, returns
+        the dimension where the concatenation happens during batching.
+
+        By default, all attribute names containing "index" will be
+        concatenated on axis 1, e.g. `edge_index_dict`. Other attributes are
+        concatenated on axis 0,
+        e.g. node features.
+
+        Args:
+            key: Name of the attribute on which change the default
+                concatenation dimension while using batching.
+
+        Returns:
+            The dimension where concatenation will happen when batching.
+        """
+        if "index" in key:
+            return 1
+        return 0
+
+    def __inc__(self, key: str, type_key: str, *args, **kwargs) -> Union[int, None]:
+        """
```

Comment:
only add `__inc__` and `__cat_dim__` if `HeteroGraphData` supports batching.
I don't see any tests with batching though, so you may want to either test batching with hetero graphs or remove that from the current PR and indicate the user that batching does not exist yet for `HeteroGraphData`